### ISSUE TITLE
set_mass

### ISSUE THREAD

**Issue by git4lei** [libAtoms/JuLIP.jl#102]
Dear Christoph, there is a small bug at atoms.jl, at line 127,
function set_masses!(at::Atoms{T}, X::Vector{T})  where T
the second argument seems to be M::Vector{T}.

Lei

**Comment by cortner** [comment]
thank you for spotting this. apparently I never used it :). 

This is of course very quick to fix, but I wondered whether you would be at all interested in making a pull-request for this fix and become an official contributor to JuLIP? It is a bit of work the first time. But once you get used to it, it becomes quite straightforward. I can take you through it. But do let me know if you prefer I do it myself, it is no trouble at all.

**Comment by cortner** [comment]
The instructions in the Julia manual are actually quite good,
```
https://docs.julialang.org/en/stable/manual/packages/#Making-changes-to-an-existing-package-1
```

But here is a slightly longer version of their "executive summary":
1. checkout the master branch, i.e., the latest development version of `JuLIP`
```
Pkg.checkout("JuLIP")     
```
2. make sure your bug is still a bug and hasn't been fixed already (I haven't fixed it, so it is still a bug)
3. go to the directory where the package is installed and checkout a new git branch in which you can make the changes:
```
cd(Pkg.dir("JuLIP"))
;git checkout -b myfixes
```
you could also for this from the command line instead of the Julia REPL:
```
cd ~/.julia/v0.6/JuLIP
git checkout -b myfixes
```
4. Edit the code
5. Make sure all tests pass (if they don't then let me know since most likely it is my problem, and not yours) may calling `Pkg.test("Images")` from the REPL
6. Commit the changes to your local git tree, either from the REPL:
```
;git commit -a -m "Bugfix for #102"  
```
or from the bash 
```
git commit -a -m "Bugfix for #102"  
```
7. Make a pull-request to JuLIP. This can be done via standard git tools, but the first time it is easier if you use Julia's package manager:
```
using PkgDev
PkgDev.submit("JuLIP")
```
This should create your own fork of JuLIP in your own GitHub account, and then open a website where you can create the pull-request. A pull-request is basically a message to me asking me to accept your changes into my own original JuLIP (i.e. to "pull" the changes from your branch into mine) I will then review the changes and accept or reject the pull-request, or ask you to make other changes as well.

**Comment by git4lei** [comment]
Thanks a lot for the detailed instruction. However, I have problems at Step 1, so maybe you or others are working on it?

Pkg.checkout("JuLIP")
INFO: Checking out JuLIP master...
ERROR: JuLIP is dirty, bailing
Stacktrace:
 [1] (::Base.Pkg.Entry.##16#18{String,String,Bool,Bool})(::Base.LibGit2.GitRepo) at ./pkg/entry.jl:228
 [2] transact(::Base.Pkg.Entry.##16#18{String,String,Bool,Bool}, ::Base.LibGit2.GitRepo) at ./libgit2/libgit2.jl:882
 [3] with(::Base.Pkg.Entry.##15#17{String,String,Bool,Bool}, ::Base.LibGit2.GitRepo) at ./libgit2/types.jl:608
 [4] checkout(::String, ::String, ::Bool, ::Bool) at ./pkg/entry.jl:226
 [5] (::Base.Pkg.Dir.##3#6{Array{Any,1},Base.Pkg.Entry.#checkout,Tuple{String,String,Bool,Bool}})() at ./pkg/dir.jl:33
 [6] cd(::Base.Pkg.Dir.##3#6{Array{Any,1},Base.Pkg.Entry.#checkout,Tuple{String,String,Bool,Bool}}, ::String) at ./file.jl:70
 [7] withenv(::Base.Pkg.Dir.##2#5{Array{Any,1},Base.Pkg.Entry.#checkout,Tuple{String,String,Bool,Bool},String}, ::Pair{String,String}, ::Vararg{Pair{String,String},N} where N) at ./env.jl:157
 [8] #cd#1(::Array{Any,1}, ::Function, ::Function, ::String, ::Vararg{Any,N} where N) at ./pkg/dir.jl:32
 [9] #checkout#1(::Bool, ::Bool, ::Function, ::String, ::String) at ./pkg/pkg.jl:188
 [10] checkout(::String) at ./pkg/pkg.jl:188

**Comment by cortner** [comment]
that means you've already modified the JuLIP repository. Form the bash 
```
cd ~/.julia/v0.6/JuLIP
git reset --hard
git checkout -b myfixes
```
then continue with 4

**Comment by cortner** [comment]
closed via #104
